build: use c++20 for all C++ files, not just those that use the seastar flags

avikivity opened the pull request:
A few source files (like those generated by antlr) don't build with seastar,
and so don't inherit all of its flags. They then use the compiler default dialect,
not C++20. With gcc that's just fine, since gcc supports concepts in earlier dialects,
but clang requires C++20.

Fix by forcing --std=c++20 for all files.

avikivity commented:
This somehow broke the build with gcc, so I dequeued it.

nyh commented:
On Mon, Oct 12, 2020 at 9:44 AM Avi Kivity <notifications@github.com> wrote:

> This somehow broke the build with gcc, so I dequeued it.
>
Interesting. Like for all your patches, I compile-tested this patch and it
didn't seem to break anything. But I think I did not try to build all the
tests, so maybe it broke one of them.

avikivity commented:
yes, it broke view_build_test.

avikivity commented:
Fixed. Changed to use --std=gnu++20.

I couldn't figure out from the error message what extension we were using. Generally I prefer not to use extensions unless strictly needed.
